inbm/cloudadapter-agent/proto/inbc-mock/inbc-mock.go

```diff
@@ -0,0 +1,114 @@
+package main
+
+import (
+	"log"
+	"net"
+	"time"
+
+	pb "inbc-mock/pb"
+
+	"github.com/google/uuid"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+)
+
+// server is used to implement inbs.INBSSBServiceServer.
+type server struct {
+	pb.UnimplementedINBSSBServiceServer
+}
+
+func authStreamInterceptor(srv interface{}, stream grpc.ServerStream, info *grpc.StreamServerInfo, handler grpc.StreamHandler) error {
+	md, ok := metadata.FromIncomingContext(stream.Context())
+	if !ok {
+		log.Print("Missing metadata from client")
+		return status.Errorf(codes.InvalidArgument, "Missing metadata")
+	}
+
+	token, ok := md["token"]
+	if !ok || len(token) == 0 || token[0] != "good_token" { // Replace with your expected token.
+		log.Print("Invalid token given by client: " + token[0])
+		return status.Errorf(codes.Unauthenticated, "invalid token")
```

Comment:
**<h2>$\color{#15F4EE}Miscellaneous$</h2>**
**Suggestion:** Consider implementing a more secure token validation mechanism. The current implementation uses a hardcoded 'good_token' string, which is not secure for production environments. Utilize environment variables or a secure vault service for token management.
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
 token, ok := md["token"]
 if !ok || len(token) == 0 || !validateToken(token[0]) {
     log.Print("Invalid token given by client: " + token[0])
     return status.Errorf(codes.Unauthenticated, "invalid token")
 }

 func validateToken(token string) bool {
     // Placeholder for token validation logic
     return false
 }
```

Reply:
this is a test client